backend/attachRendererFiber.js

```diff
@@ -148,11 +155,20 @@ function attachRendererFiber(hook: Hook, rid: string, renderer: ReactRenderer):
     var actualStartTime = null;
     var treeBaseDuration = null;
 
+    var resolvedType = type;
+    if (typeof type === 'object' && type !== null) {
+      if (typeof type.then === 'function') {
+        resolvedType = type._reactResult;
+      }
+    }
+
     switch (fiber.tag) {
       case FunctionalComponent:
+      case FunctionalComponentLazy:
       case ClassComponent:
+      case ClassComponentLazy:
         nodeType = 'Composite';
-        name = getDisplayName(fiber.type);
+        name = getDisplayName(resolvedType);
```

Comment:
These shouldn't appear in the tree before type is resolved. So I'm assuming it is here.